### PR TITLE
Vault migration pass

### PR DESCRIPTION
**Pull request by apentori**

### CODE REVIEW

File: vault/migration-bitwarden.sh [hunk 1]
```diff
@@ -33,11 +33,23 @@ if [[ -n $3 ]]; then
...
 elif [[ $TYPE == "pwd" ]]; then
   critera=".login | {username: .username, password: .password}"
 fi
 
 echo "Writing Bitwarden ${TYPE} of ${BITWARDEN_PATH} into ${VAULT} of ${VAULT_PATH}"
-bw get item ${BITWARDEN_PATH} | jq -r "${critera}" | vault kv put -mount=${VAULT} $VAULT_PATH -
+echo ${object} | jq -r "${critera}" | vault kv put -mount=${VAULT} $VAULT_PATH -
+# Update notes
+timestamp=$(date  +%F)
+jq_note=".notes=\"${notes} Migrated to Vault - ${timestamp}\""
```
Comment (jakubgs) [inline vault/migration-bitwarden.sh:54]: I was thinking more about adding a custom field or two like `vault-migartion` for timestamp and `vault-path`.

**Review by jakubgs — APPROVED** [review]

File: vault/migration-pass.sh [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  echo "Please set the varaible $VAULT_TOKEN"
+  exit 1
+fi
+if [[ -z $VAULT_ADDR ]]; then
+  echo "Please set the varaible $VAULT_ADDR"
+  exit 1
+fi
+
+if [[ -z "${*}" ]]; then
+    echo "Usage: migration-pass.sh <bitwarden_path> <vault_path>"
```
Comment (yakimant) [inline vault/migration-pass.sh]: bitwarden?

File: vault/migration-pass.sh [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    echo
+    echo " - password_store_path: path of the secret in bitwarden"
```
Comment (yakimant) [inline vault/migration-pass.sh]: bitwarden?

**Review by yakimant — APPROVED** [review]
I wonder if it could be an issue that smth copyied from bw/pass to vault, but kept in the previous place. This might cause having the same thing in two places and eventually having them unsynced.

I'm not saying we need to automatically remove secrets from bw/pass while running the scripts, but we need to keep it in mind. Maybe doing some cleanup afterwards or in any other way.

Also, I don't see `migration-pass.sh` doing actual migration, am I blind?

**Comment by apentori** [comment]
The `migration-pass` has been commited by mistake, it's not finished and doesn't migrate anything yet, you are not blind !

**Comment by jakubgs** [comment]
>This might cause having the same thing in two places and eventually having them unsynced.

@yakimant that's why I suggested adding the field to migrated BitWarden secrets, so we can identify stuff that was moved.
We could also add a check to `ansible/lookup_plugins/bitwarden.py` which will cause a failure if migration attirbutes are set.

**Comment by apentori** [comment]
> @yakimant that's why I suggested adding the field to migrated BitWarden secrets, so we can identify stuff that was moved.
> We could also add a check to `ansible/lookup_plugins/bitwarden.py` which will cause a failure if migration attirbutes are set.

Condition  acceded here:

https://github.com/status-im/infra-template/pull/8

Script to migrate secret from pass finalized

**Pull request merged**